pkg/frontend/openshiftcluster_putorpatch_test.go

```diff
@@ -53,6 +54,299 @@ func expectAsyncOperationDocumentCreate(asyncOperations *mock_database.MockAsync
+
+	mockSubID := "00000000-0000-0000-0000-000000000000"
+
+	type test struct {
+		name           string
+		resourceID     string
+		request        func(*admin.OpenShiftCluster)
+		isPatch        bool
+		mocks          func(*test, *mock_database.MockAsyncOperations, *mock_database.MockOpenShiftClusters, *mock_clusterdata.MockOpenShiftClusterEnricher)
+		wantStatusCode int
+		wantResponse   func(*test) *admin.OpenShiftCluster
+		wantAsync      bool
+		wantError      string
+	}
+
+	for _, tt := range []*test{
+		{
+			name:       "patch with empty request",
+			resourceID: fmt.Sprintf("/subscriptions/%s/resourcegroups/resourceGroup/providers/Microsoft.RedHatOpenShift/openshiftClusters/resourceName", mockSubID),
+			request: func(oc *admin.OpenShiftCluster) {
+			},
+			isPatch: true,
+			mocks: func(tt *test, asyncOperations *mock_database.MockAsyncOperations, openShiftClusters *mock_database.MockOpenShiftClusters, enricher *mock_clusterdata.MockOpenShiftClusterEnricher) {
+				currentClusterdoc := &api.OpenShiftClusterDocument{
+					Key: strings.ToLower(tt.resourceID),
+					OpenShiftCluster: &api.OpenShiftCluster{
+						ID:   tt.resourceID,
+						Type: "Microsoft.RedHatOpenShift/openshiftClusters",
+						Tags: map[string]string{"tag": "will-be-kept"},
+						Properties: api.OpenShiftClusterProperties{
+							ProvisioningState: api.ProvisioningStateSucceeded,
+						},
+					},
+				}
+
+				openShiftClusters.EXPECT().
+					Get(gomock.Any(), strings.ToLower(tt.resourceID)).
+					Return(currentClusterdoc, nil)
+
+				enricher.EXPECT().Enrich(gomock.Any(), currentClusterdoc.OpenShiftCluster)
+				expectAsyncOperationDocumentCreate(asyncOperations, strings.ToLower(tt.resourceID), api.ProvisioningStateAdminUpdating)
+
+				clusterdoc := &api.OpenShiftClusterDocument{
+					Key: strings.ToLower(tt.resourceID),
+					OpenShiftCluster: &api.OpenShiftCluster{
+						ID:   tt.resourceID,
+						Type: "Microsoft.RedHatOpenShift/openshiftClusters",
+						Tags: map[string]string{"tag": "will-be-kept"},
+						Properties: api.OpenShiftClusterProperties{
+							ProvisioningState:     api.ProvisioningStateAdminUpdating,
+							LastProvisioningState: api.ProvisioningStateSucceeded,
+						},
+					},
+				}
+
+				openShiftClusters.EXPECT().
+					Update(gomock.Any(), (*matcher.OpenShiftClusterDocument)(clusterdoc)).
+					Return(clusterdoc, nil)
+			},
+			wantAsync:      true,
+			wantStatusCode: http.StatusOK,
+			wantResponse: func(tt *test) *admin.OpenShiftCluster {
+				return &admin.OpenShiftCluster{
+					ID:   tt.resourceID,
+					Type: "Microsoft.RedHatOpenShift/openshiftClusters",
+					Tags: map[string]string{"tag": "will-be-kept"},
+					Properties: admin.OpenShiftClusterProperties{
+						ProvisioningState:     admin.ProvisioningStateAdminUpdating,
+						LastProvisioningState: admin.ProvisioningStateSucceeded,
+					},
+				}
+			},
+		},
+		{
+			name:       "create a new cluster with registry profile set, should ignore registry profile",
```

Comment:
This is confusing. Admin API can't create clusters. So this test is confusing. Admin can do only upgrades/updates so tests should be doing this. For somebody who gonna read this, this might give the wrong impression despite the fact that it tests right functionality.